File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed) => speed switch
    {
        0   => 0,
        < 5 => 1,
        < 9 => 0.9,
        9   => 0.8,
        10  => 0.77,
        _   => throw new InvalidOperationException("An invalid speed value."),
    };
    
    public static double ProductionRatePerHour(int speed) => 221.0 * speed * SuccessRate(speed);

    public static int WorkingItemsPerMinute(int speed) => (int)(ProductionRatePerHour(speed) / 60);
}

File: solutions/fsharp/cars-assemble/1/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    match speed with
    | s when 1 <= s && s <= 4 -> 1.0
    | s when 5 <= s && s <= 8 -> 0.9
    | 9 -> 0.8
    | 10 -> 0.77
    | _ -> 0

let productionRatePerHour (speed: int) : float =
    221.0 * float speed * successRate (speed)

let workingItemsPerMinute (speed: int) : int =
    int (productionRatePerHour speed / 60.0)
